src/test/regress/sql/limit.sql

```diff
@@ -147,3 +147,16 @@ select sum(tenthous) as s1, sum(tenthous) + random()*0 as s2
 
 select sum(tenthous) as s1, sum(tenthous) + random()*0 as s2
   from tenk1 group by thousand order by thousand limit 3;
+
+-- Check the operation of the parameterized Limit in a subquery with a volatile function
+-- start_ignore
+drop table if exists limit_tbl;
+-- end_ignore
+create table limit_tbl(i int) distributed by (i);
+insert into limit_tbl select * from generate_series(1, 3) i;
+
+explain (verbose, costs off)
+select (select a + random()*0 from generate_series(1, 4) a limit 1 offset limit_tbl.i) as r from limit_tbl order by r;
+select (select a + random()*0 from generate_series(1, 4) a limit 1 offset limit_tbl.i) as r from limit_tbl order by r;
```

Comment:
Why all the hassle here with multiplying the random by zero? Wouldn't it be easier to use a volatile function that returns its argument?
```suggestion
create function f(i int) returns int language plpgsql as $$ begin return i; end; $$;
explain (verbose, costs off)
select (select f(a) from generate_series(1, 4) a limit 1 offset limit_tbl.i) as r from limit_tbl order by r;
select (select f(a) from generate_series(1, 4) a limit 1 offset limit_tbl.i) as r from limit_tbl order by r;
```